src/sentry/workflow_engine/models/data_condition.py

```diff
@@ -103,3 +106,18 @@ def evaluate_value(self, value: float | int) -> DataConditionResult:
             return self.get_condition_result()
 
         return None
+
+
+class DetectorWorkflowCondition(abc.ABC):
```

Comment:
let's chat about this class in office hours, i'd like to understand the reason we need it vs using a data condition itself.